identity/tests/db_queries.rs

```diff
@@ -0,0 +1,336 @@
+use std::env::set_var;
+use std::sync::Arc;
+
+use chrono::NaiveDate;
+use diesel::r2d2::{ConnectionManager, Pool};
+use diesel::{prelude::*, result::Error};
+use envconfig::Envconfig;
+
+use identity::config::Configuration;
+use identity::db::schema::{users::dsl::users, users_roles::dsl::users_roles};
+use identity::db::{get_db_pool, models::*, queries};
+
+mod sample_data;
+
+#[macro_use]
+extern crate diesel_migrations;
+
+embed_migrations!("migrations/");
+
+struct DbTestContext {
+    connection_url: String,
+    db_name: String,
+}
+
+impl DbTestContext {
+    fn new(connection_url: String, db_name: String) -> Self {
```

Comment:
If you use some Result as return type you can use the `?` operator which would simplify some stuff.
I'd recommend:
```rust
Result<Self, Box<dyn std::error::Error>>
```